test/msgpack.cpp

```diff
@@ -11,7 +11,7 @@ std::vector<char> msgpack_buffer(const T& src)
     msgpack::pack(buffer, src);
     buffer.seekg(0);
     std::string str = buffer.str();
-    return std::vector<char>(str.data(), str.data() + str.size());
+    return std::vector<char>(str.data(), str.data() + str.size()); // NOLINT
```

Comment:
Why does this use `NOLINT`?

Reply:
I tried rewriting using a brace initializer list, but then I get a tidy error that the local variable `str` should not be used within the initializer list when returning from the function. I had a workaround commit where I just replaced `str` with `buffer.str()`, but then the msgpack test failed.